internal/credential/service_library.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package credential
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/util"
+)
+
+// VaultLibraryRepository defines the interface expected
+// to get the total number of credential libraries and deleted ids.
+type VaultLibraryRepository interface {
+	EstimatedLibraryCount(context.Context) (int, error)
+	EstimatedSSHCertificateLibraryCount(context.Context) (int, error)
+	ListDeletedLibraryIds(context.Context, time.Time, ...Option) ([]string, error)
+	ListDeletedSSHCertificateLibraryIds(context.Context, time.Time, ...Option) ([]string, error)
+	ListCredentialLibraries(context.Context, string, ...Option) ([]Library, error)
+	ListSSHCertificateCredentialLibraries(context.Context, string, ...Option) ([]Library, error)
+}
```

Comment:
This doesn't seem like a "repository" to me, even though these are all methods that exist on the vault.Repository, this combination of methods by themselves does not seem like a repository.

In addition, this exposes details of `vault` into `credential`. I am thinking it might make sense to instead create an interface like:

```go
type SubtypeLibraryService interface {
    EstimatedCount(context.Context) (int, error)
    ListDeletedIds(context.Context, time.Time, ...Option) ([]string, error)
    List(context.Context, string, ...Option) ([]Library, error)
}
```

Then in `vault` add a service that implements this interface. This allows for vault to manage combining SSHCertificateLibraries with GenericLibraries. Then the `credential.LibraryService` can receive a `[]SubtypeLibraryService` that can be expanded if/when there are other non-vault credential libraries.

Reply:
That's a nice idea, let me try it!

Reply:
I liked this, and I think it's also a pattern that we could in theory apply to more resources, but I don't think it's worth the extra indirection. Please take a look!